retrolambda/src/main/java/net/orfjackal/retrolambda/lambdas/BackportLambdaInvocations.java

```diff
@@ -116,9 +121,55 @@ private void backportLambda(String invokedName, Type invokedType, Handle bsm, Ob
             Handle implMethod = (Handle) bsmArgs[1];
             Handle accessMethod = getLambdaAccessMethod(implMethod);
 
-            LambdaFactoryMethod factory = LambdaReifier.reifyLambdaClass(implMethod, accessMethod,
+            LambdaClassInfo info = LambdaReifier.reifyLambdaClass(implMethod, accessMethod,
                     invoker, invokedName, invokedType, bsm, bsmArgs);
-            super.visitMethodInsn(INVOKESTATIC, factory.getOwner(), factory.getName(), factory.getDesc(), false);
+
+            if (info.isStateless()) {
+                super.visitFieldInsn(GETSTATIC, info.getLambdaClass(), info.getReferenceName(),
+                        info.getReferenceDesc());
+                return;
+            }
+
+            // Capturing method references emit 3 useless bytecodes immediately before the invokedynamic
+            // instruction which we are currently visiting:
+            //   3: aload_0
+            //   4: dup
+            //   5: invokevirtual #3      // Method java/lang/Object.getClass:()Ljava/lang/Class;
+            //   8: pop
+            //   9: invokedynamic #4,  0  // InvokeDynamic #0:call:(Ljava/lang/String;)Ljava/util/concurrent/Callable;
+            // Attempt to match whether these (4, 5, and 8) are the three preceding bytecodes and remove them if so.
```

Comment:
They screw up the calculation of how far "up" to place the `new` and `dup` instructions because they are between the argument loads and `invokedynamic` call. Instead of removing I can just walk backwards through the instructions until N loads are seen (where N is the number of arguments required).